### DISCUSSION TITLE
How to use TFT_eSPI library for 16 bit colour rendering?

### DISCUSSION THREAD

**Discussion by reflectingme** [Bodmer/TFT_eSPI#2478] [Compatible displays and setup files]
OK, werid issue here, probably down to my ignorance but wondering if anyone can point me in the right direction please.
I am using the LilyGo T-Embed ESP32S3 device which is working almost as I wish, however when using the TFT_eSPI library, the colors displayed either as backgrounds, text or drawn objects are all incorrect.
For example, the code:

`tft.fillScreen(TFT_RED);`

actually fills the screen with blue!! and...

```
tft.setTextColor(TFT_YELLOW, TFT_BLACK);
tft.drawString("testing yellow", 10, 100, 4);
```
... renders teal coloured text and NOT yellow!!

Now, I am sure it is something to do with the color bit rendering and type, which for all the other T-Display boards I have is RGB565 and with the very same TFT_eSPI library works fine.  These boards are 8 bit colour but I think the T-Embed is 16 bit.

How do I use the TFT_eSPI library for 16 bit colour?

**Comment by Bodmer** [comment]
Red and Blue pixels are swapped. This is at a hardware level in the display screen. Add ONE of the following lines in your setup file.
```
#define TFT_RGB_ORDER TFT_RGB  // Colour order Red-Green-Blue
#define TFT_RGB_ORDER TFT_BGR  // Colour order Blue-Green-Red
```
